docs/book/rest-api.md

```diff
@@ -1400,11 +1400,24 @@ If the default decision (defaulting to `/system/main`) is undefined, the server
 
 ### Execute an Ad-hoc Query
 
+Execute an ad-hoc query and return bindings for variables found in the query.
+
 ```
 GET /v1/query
 ```
 
-Execute an ad-hoc query and return bindings for variables found in the query.
+For queries that have large JSON values it is recommended to use the `POST` method with the query included as the `POST` body
+
+```
+POST /v1/query HTTP/1.1
+Content-Type: application/json
+```
+
+```json
+{
+  "query": ...
```

Comment:
Nit: can you include an example query here? There's a query specified in the example below that you could copy.

Reply:
Done.